geps/gep-2648/index.md

```diff
@@ -0,0 +1,415 @@
+
+// PolicyConditionReason is a reason for a policy condition.
+type PolicyConditionReason string
+
+const (
+  // PolicyConditionAccepted indicates whether the policy has been accepted or rejected
+  // by a targeted resource, and why.
+  //
+  // Possible reasons for this condition to be True are:
+  //
+  // * "Accepted"
+  //
+  // Possible reasons for this condition to be False are:
+  //
+  // * "Conflicted"
+  // * "Invalid"
+  // * "TargetNotFound"
+  //
+  PolicyConditionAccepted PolicyConditionType = "Accepted"
+
+  // PolicyReasonAccepted is used with the "Accepted" condition when the policy has been
+  // accepted by the targeted resource.
+  PolicyReasonAccepted PolicyConditionReason = "Accepted"
+
+  // PolicyReasonConflicted is used with the "Accepted" condition when the policy has not
+  // been accepted by a targeted resource because there is another policy that targets the same
+  // resource and a merge is not possible.
+  PolicyReasonConflicted PolicyConditionReason = "Conflicted"
+
+  // PolicyReasonInvalid is used with the "Accepted" condition when the policy is syntactically
+  // or semantically invalid.
+  PolicyReasonInvalid PolicyConditionReason = "Invalid"
+
+  // PolicyReasonTargetNotFound is used with the "Accepted" condition when the policy is attached to
+  // an invalid target resource
+  PolicyReasonTargetNotFound PolicyConditionReason = "TargetNotFound"
+)
+```
+
+#### On targeted resources
+
+Implementations that use Direct Policy objects SHOULD put a Condition into
+`status.Conditions` of any objects affected by a Direct Policy, if that field
+is present.
+
+If they do, that Condition MUST have a `type` ending in `PolicyAffected` (like
+`gateway.networking.k8s.io/PolicyAffected`),
+and have the optional `observedGeneration` field kept up to date when the `spec`
+of the Policy-attached object changes.
+
+Implementations SHOULD use their own unique domain prefix for this Condition
+`type` - it is recommended that implementations use the same domain as in the
+`controllerName` field on GatewayClass (or some other implementation-unique
+domain for implementations that do not use GatewayClass).
+
+For objects that do _not_ have a `status.Conditions` field available (`Secret`
+is a good example), that object SHOULD instead have an annotation of
+`gateway.networking.k8s.io/PolicyAffected: true` (or with an
+implementation-specific domain prefix) added instead.
+
+Because these Conditions or annotations are namespaced per-implementation,
+implementations SHOULD:
+
+- Add the Condition or annotation if an object is policy affected when it is not
+  already present
+- Remove the Condition or annotation when the last policy object stops referencing
+  the targeted object.
+
+### Other Status designs
+
+This section contains other recommendations for status designs. Note that this
+is a SHOULD rather than a MUST, as this design is still not final.
+
+
+#### Standard status struct
+
+Status: Experimental
```

Comment:
Why does this need a separate status?

Reply:
This is an attempt to codify what we're doing in BackendTLSPolicy into this GEP, that's all.